src/routes/MapExplorer/MapExplorer.js

```diff
@@ -0,0 +1,229 @@
+import React from 'react'
```

Comment:
This all LGTM! And wondering what this component would be like if implemented with hooks? Like would it be more or less legible?

Reply:
as soon as components get complicated i find it hard to stick with the pure function component